# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/Sweeping.cpp
# Commit: 59ae32a4a0a3
# Full Hash: 59ae32a4a0a3be6160a757099e38bbc5f52db4b1
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2022-03-01 16:17:59
# Regressor Bug: 1755725
# File Overlap Count: 1
# Description:
#   Bug 1755725 - Part 1: Factor out crossZoneWrappers checks and do them before we calculate sweep groups r=sfink
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D139835
# ==============================================================================

diff -r ae499200e145 -r 59ae32a4a0a3 js/src/gc/Sweeping.cpp
--- a/js/src/gc/Sweeping.cpp	Tue Mar 01 08:27:54 2022 +0000
+++ b/js/src/gc/Sweeping.cpp	Tue Mar 01 08:28:41 2022 +0000
@@ -628,6 +628,12 @@
 }
 
 bool Zone::findSweepGroupEdges(Zone* atomsZone) {
+#ifdef DEBUG
+  if (FinalizationObservers* observers = finalizationObservers()) {
+    observers->checkTables();
+  }
+#endif
+
   // Any zone may have a pointer to an atom in the atoms zone, and these aren't
   // in the cross compartment map.
   if (atomsZone->wasGCStarted() && !addSweepGroupEdgeTo(atomsZone)) {